src/tokenized-buffer.coffee

```diff
@@ -126,9 +125,17 @@ class TokenizedBuffer extends Model
     @disposables.add(@configSubscriptions)
 
     @retokenizeLines()
-    @packageManager.triggerActivationHook("#{grammar.packageName}:grammar-used")
+
     @emitter.emit 'did-change-grammar', grammar
 
+    # Delay this to the next tick to ensure whoever created the buffer has the
+    # change to listen for this event before we send it.
+    process.nextTick =>
```

Comment:
@atom/core Thoughts on this? Alternatively, consumers could assume this happens synchronously and Deal With the fact that they'll miss this. But I'd rather not build in more synchronous assumptions.